packages/bundle-size/src/utils/collectLocalReport.js

```diff
@@ -0,0 +1,47 @@
+const fs = require('fs').promises;
+const glob = require('glob');
+const path = require('path');
+const { findGitRoot, findPackageRoot } = require('workspace-tools');
+
+/** @typedef {import('../utils/buildFixture').BuildResult} BuildResult */
+
+/** @typedef {BuildResult & { packageName: string }} BundleSizeReportEntry */
+/** @typedef {BundleSizeReportEntry[]} BundleSizeReport */
+
+/**
+ * @param {string} reportFile
+ *
+ * @return {Promise<[string, BuildResult[]]>}
+ */
+async function readReportForPackage(reportFile) {
+  const reportFilePath = path.resolve(process.cwd(), reportFile);
+
+  const packageName = path.basename(/** @type {string}*/ (findPackageRoot(reportFilePath)));
+  const packageReport = JSON.parse((await fs.readFile(reportFilePath)).toString());
+
+  return [packageName, packageReport];
+}
+
+/**
+ * Collects all reports for packages to a single one.
+ *
+ * @return {Promise<BundleSizeReport>}
+ */
+async function collectLocalReport() {
+  /** @type {string[]} */
+  const reportFiles = glob.sync('packages/*/dist/bundle-size/bundle-size.json', {
+    cwd: /** @type {string} */ (findGitRoot(process.cwd())),
+  });
+
+  /** @type {[string, BuildResult[]][]} */
+  const reports = await Promise.all(reportFiles.map(readReportForPackage));
+
+  return reports.reduce(
```

Comment:
this reduce is not very human readable and contains lot of overrides which tends to point at code smells. 

let's say we use record api as a result from `readReportForPackages` we can write reduce like this:

```js
const reports = await Promise.all(reportFiles.map(readReportForPackage));

return reports.reduce((acc, { packageName, packageReport }) => {
    const processedReport = packageReport.map(reportEntry => ({ packageName, ...reportEntry }));

    return [...acc, ...processedReport];
  }, /** @type {BundleSizeReport} */ ([]));
}
```

Reply:
Applied in 2841356.